store/storage/rocksdb/iterator.go

```diff
@@ -0,0 +1,129 @@
+package rocksdb
+
+import (
+	"bytes"
+
+	"cosmossdk.io/store/v2"
+	"github.com/linxGnu/grocksdb"
+)
+
+var _ store.Iterator = (*iterator)(nil)
+
+type iterator struct {
+	source             *grocksdb.Iterator
+	prefix, start, end []byte
+	reverse            bool
+	invalid            bool
+}
+
+func newRocksDBIterator(source *grocksdb.Iterator, prefix, start, end []byte, reverse bool) *iterator {
+	if reverse {
+		if end == nil {
+			source.SeekToLast()
+		} else {
+			source.Seek(end)
+
+			if source.Valid() {
+				eoaKey := copyAndFreeSlice(source.Key()) // end or after key
```

Comment:
A prime candidate for the readOnly slice instead of this expensive copy on every comparison